example/helper.rb

```diff
@@ -17,3 +17,25 @@ def info(msg)
     puts "[Stream #{@id}]: #{msg}"
   end
 end
+
+def send_promises(stream)
+  push_streams = []
+
+  # send 10 promises
+  10.times do |i|
+    sleep 1
+    puts 'sending push'
+
+    head = {
+      ':status' => '200',
+      ':path'   => "/other_resource/#{i}",
+      'content-type' => 'text/plain',
+    }
+
+    stream.promise(head) do |push|
+      push.headers(head)
```

Comment:
Without `push.headers(head)`, sending data raises an `InternalError`.

```
Sent frame: {:type=>:push_promise, :flags=>[:end_headers], :stream=>1, 
:promise_stream=>2, :payload=>[[":status", "200"], [":path", "/other_resource/0"], 
    ["content-type", "text/plain"]]}

Sent frame: {:type=>:headers, :flags=>[:end_headers],
 :payload=>[[":status", "200"], [":path", "/other_resource/0"], ["content-type", "text/plain"]],
 :stream=>2}
````

It seems that `stream.promise(head)` sends the promise on the "original" stream. `push.headers(head)` appears to send the headers on the "push" stream.